### PR TITLE
Henry/addstaff

### PR DESCRIPTION
**Pull request by chanhenry54**
Contains commits for both class creation and adding staff. Still needs to implement connecting to the back-end through the API. Make sure to `yarn install` for access to react-tag-autocomplete package. I believe Ronak said he could help with that? The Staff methods just need to update the instructorList property of the course.
Methods that need connection to back-end:
- ClassCreation/index.jsx: handleClassCreation() --> done
- Staff/index.jsx: handleStaffInvite() --> done
- Staff/index.jsx: removeInstructor() --> done
- Staff/index.jsx: renderClassInstructors() --> done

### CODE REVIEW

File: client/src/pages/ClassCreation/index.jsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    /* Main function for handling class creation */
+    const handleClassCreation = async (event) => {
+        event.preventDefault();
+        if (titleValid && descValid) {
+            const { title, description, term } = event.target.elements;
+            const tags = selectedTags.tags;
+
+            /* try to create a course in database */
+            try {
+                // TODO: use API to create course, generate a courseID, add user to instructorList property of course
```
Comment (daniel-d-truong) [inline client/src/pages/ClassCreation/index.jsx:43]: Assign Ronak and point him to this line to get data dynamically

File: client/src/pages/ClassCreation/index.jsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                            <p>Course Title:</p>
+                            <input name="title" placeholder="Ex: CSE 111 - History of LeetCode" onBlur={handleTitleChange} style={{ backgroundColor: titleBgColor }} />
+                        </div>
+                        <div className="flex-row">
+                            <p>Description:</p>
+                            <textarea name="description" placeholder="Description" onBlur={handleDescChange} style={{ backgroundColor: descBgColor }}></textarea>
+                        </div>
+                        <div className="flex-row">
+                            <p>Course Term:</p>
+                            <select name="term">
```
Comment (daniel-d-truong) [inline client/src/pages/ClassCreation/index.jsx:134]: Not sure if we want to dynamically render the available list of terms (we could also provide two dropdown options, with one being the season and the second being the year). This is a low priority though so we can deal with this later.

File: client/src/pages/Staff/index.jsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    /* Function for handling inviting staff */
+    const handleStaffInvite = (event) => {
+        event.preventDefault();
+        const { email } = event.target.elements;
+
+        // try to add staff to course in database
+        try {
+            // TODO: API stuff, check if already in database
```
Comment (daniel-d-truong) [inline client/src/pages/Staff/index.jsx:23]: Assign Ronak to this line as well

**Comment by daniel-d-truong** [comment]
One thing I noticed:

image

Buttons are a bit weird and we can update the navbar.

File: backend/controllers/courseController.js [hunk 4]
```diff
@@ -188,17 +190,49 @@ exports.deleteCourse = async (req, res) => {
     };
 };
 
+// Gets all the course users
+exports.getCourseUsers = async (req, res) => {
+    const courseUUID = req.params.courseId;
+    if (!courseUUID) {
+        res.status(422).json({
+            status: 422,
+            error: "Missing paramater: courseUUID"
```
Comment (chanhenry54) [inline backend/controllers/courseController.js:199]: parameter

File: backend/controllers/courseController.js [hunk 5]
```diff
@@ -188,17 +190,49 @@ exports.deleteCourse = async (req, res) => {
...
+
 // Sends an email to the passed in user (defaults to student)
 exports.sendEmail = async (req, res) => {
+
     const courseUUID = req.params.courseId;
     if (!courseUUID) {
         res.status(422).json({
             status: 422,
-            error: "Missing parameter: courseUUID"
+            error: "Missing paramater: courseUUID"
```
Comment (chanhenry54) [inline backend/controllers/courseController.js:230]: parameter

File: backend/controllers/courseController.js [hunk 6]
```diff
@@ -240,5 +274,43 @@ exports.sendEmail = async (req, res) => {
...
+
+
+exports.removeUser = async (req, res) => {
+    const courseUUID = req.params.courseId;
+    const userUUID = req.params.userId;
+    
+    if (!courseUUID || !userUUID) {
+        res.status(422).json({
+            status: 422,
+            error: "Missing paramater: courseUUID or userUUID"
```
Comment (chanhenry54) [inline backend/controllers/courseController.js:289]: parameter

File: backend/models/User.js [hunk 7]
```diff
@@ -132,6 +132,35 @@ class User {
...
+    removeInstructorCourse = async (courseId) => {
+        await this.updateInstructorCourses();
+        
+        if (this.props.instructorCourseList.indexOf(courseId) >= 0) {
+            this.props.instructorCourseList.splice(this.props.instructorCourseList.indexOf(courseId), 1);
+            await this.push();
+        }
+    }
+
+    removeStduentCourse = async (courseId) => {
```
Comment (chanhenry54) [inline backend/models/User.js:155]: removeStudentCourse

**Pull request merged**